components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,173 @@
+        tooltip.style.top = Math.max((targetRectCenterY - (tooltip.clientHeight/2)), tooltipMargin) + 'px'
+        tooltipRect = tooltip.getBoundingClientRect()
+        tooltipPointer.style.bottom = 'auto'
+        tooltipPointer.style.top = Math.max((((pointerWidth * Math.sqrt(2))/2) - (pointerWidth/2)), (targetRectCenterY - tooltipRect.top - window.scrollY - (pointerWidth/2))) + 'px'
+        tooltipPointerLeft = - (pointerWidth/2) + 'px'
+      } else {
+        tooltip.style.top = targetRect.bottom + pointerGap + (pointerWidth/2) + window.scrollY + 'px'
+        tooltipPointer.style.top = 'auto'
+        tooltipPointer.style.bottom = tooltip.clientHeight - (pointerWidth/2) + 'px'
+      }
+    } else {
+      tooltip.style.top = targetRect.top - tooltip.clientHeight + window.scrollY - pointerGap - (pointerWidth/2) + 'px'
+      tooltipPointer.style.bottom = - (pointerWidth/2) + 'px'
+      tooltipPointer.style.top = 'auto'
+    }
+    tooltipPointer.style.left = tooltipPointerLeft
+
+    if (tooltip.classList.contains('tooltip-hide')) {
+      tooltip.classList.remove('tooltip-hide')
+      tooltip.style.transition = 'opacity ' + tooltipDelay + 'ms linear'
+      tooltip.style.opacity = '1'
+    }
+  }
+
+  hideTooltip() {
+    const tooltip = ReactDOM.findDOMNode(this)
+    if (!tooltip.classList.contains('tooltip-hide')) {
+      tooltip.classList.add('tooltip-hide')
+      tooltip.style.transition = 'opacity 0s linear'
+      tooltip.style.opacity = '0'
+    }
+  }
+
+  onClick(evt) {
+    if (this.state.isActive) {
+      this.hideTooltip()
+    } else {
+      this.showTooltip(evt)
+    }
+
+    this.setState({ isActive: !this.state.isActive })
+  }
+
+  componentDidMount() {
+    const targetId = this.props.tooltipId
+    const target = document.getElementById(targetId)
+    if (this.props.popMethod === 'hover') {
+      target.addEventListener('mouseenter', this.showTooltip)
+      target.addEventListener('mouseleave', this.hideTooltip)
+    } else if (this.props.popMethod === 'click') {
+      target.classList.add('click-pointer')
+      target.addEventListener('click', this.onClick)
+    }
+  }
+
+  componentWillUnmount() {
+    const targetId = this.props.tooltipId
+    const target = document.getElementById(targetId)
+    target.removeEventListener('mouseenter', this.showTooltip)
+    target.removeEventListener('mouseleave', this.hideTooltip)
+
+    target.removeEventListener('click', this.onClick)
+
+  }
+
+  render() {
+    const tooltipTheme = this.props.theme
+    const body = (
+      <div className="tooltip-body">
+          {React.Children.map(this.props.children, (child) => {
+            return child.props.children
+          })}
+      </div>
+    )
+    let ttClasses = 'Tooltip tooltip-hide' + ' ' + tooltipTheme
+    if (typeof this.props.className !== 'undefined') {
+      ttClasses += ' ' + this.props.className
+    }
+    return (
+      <div className={ttClasses}>
```

Comment:
Can you please use classnames module for this? You can see other components for reference. Many of the existing components use this module.